factorial/fact.cpp

```diff
@@ -0,0 +1,36 @@
+#include<iostream>
+int fun(int n)
+{
+if(n==0)
+    {
+        return 1;
+    }
+    else
+    {
+            n=n*fun(n-1);
+    }
+}
+int main()
+{
+    int n;
+    char c;
+    std::cout<<"Enter a positive Integer to find Factorial: ";
+    std::cin>>n;
+    std::cout<<"\n";
+    if(n>=0 && n!=c)
```

Comment:
Why are you comparing integer n to an uninitialized char variable c.
This is undefined behaviour and might crash your program